### DISCUSSION TITLE
Tempoffset

### DISCUSSION THREAD

**Discussion by dippes** [arendst/Tasmota#21537] [Q&A]
Hello

My Shelly Plus 1 shows 3.3 degrees on the temp sensor but in reality the room temp. is approx. 22 degrees. temp offset is possible up to a maximum of 12.6 degrees. how do I make the difference. is this even possible?

**Comment by sfromis** [comment]
What you want to do has nothing to do with calibrating for an offset. With such a **huge** difference, you need to configure `AdcParam` correctly for the sensor, instead of trying to add a value to a broken number. The point is that you cannot assume that a broken setup will have the same deviation across temperatures. Of course, this is assuming that the hardware is not outright broken.

**Comment by dippes** [comment]
How do I find out the AdcParam for my Shelly 1 Mini 3 GEN?

I flashed it today with Tasmota and set it up, but the WIF shows a temperature of 65.5 C. The Shelly is not yet installed anywhere and is lying on the table.

**Reply by arendst** [reply]
Try this ``AdcParam1 2,5600,4700,3350``

**Comment by dippes** [comment]
Unfortunately, it didn't help much and now a temperature of 58.7 C is displayed. Room temperature approx. 23 C

**Reply by arendst** [reply]
Well, as the sensor is inside the box don't expect to read 23 C !!!

**Reply by sfromis** [reply]
Varying by device type, that sort of internal temperature is not even surprising for Shelly devices, like when I can feel a Shelly 2PM Plus being quite warm to the touch. You certainly can't use the internal sensor for ambient temperature, it is meant for protection against overheating.

**Comment by dippes** [comment]
58.7 C in the uninstalled state and without consumers is that not too much?
How much temperature can the Shelly withstand before it dies?

**Reply by sfromis** [reply]
Did you try holding it in the hand to check how warm it feels? There is no guarantee that the `AdcParam` settings is right for the device, but the value you see is not implausible. As long as it is below the boiling point, it is not likely to harm the device.

The one where I could feel the heat reports a much higher temperature than my hand would expect (without any output load), but I'm not certain that the `AdcParam` settings are correct there.

**Comment by dippes** [comment]
I measured with an infrared heat meter.

While the WIF shows 56.4 C. I measured 31.5 C on the top of the case.

The Shelly doesn't feel warmer in your hand either.

How can I adjust the temp difference?

**Reply by sfromis** [reply]
It is perfectly reasonable that the board inside is a lot hotter than the case. Not something to be "adjusted".

**Comment by arendst** [comment]
I'm researching the issue too.

For now this should be the correct parameter for the Shelly Plus 1: ``adcparam1 2,10000,10000,3350``. With that setting the temp will even be higher than before but at least the resistor values are correct.

My Shelly 1 PM mini also shows a temp of 64.3 C even with open case. Touching the sensor when power removed does not feel like that high.

I feel there is something different around the ADC channels of the shelly (using ESP32C3) and other ESP32 based devices. Investigating....

UPDATE: In below picture the sensor can be found just below the ESP32C3 chip next to the 3.3v pin so it is very much depending on the heat of this chip.
![shelly_plus_1pm_mini](https://github.com/user-attachments/assets/41e4fdd9-bc87-4e36-955f-6518bb677fa1)

**Comment by arendst** [comment]
After some more tests I think the Shelly temperature sensor is reporting a valid temperature. With the case open and powered off for an hour, then powered on it reports room temperature increasing rapidly to around 65 C which is probably the right temp next to the ESP32C3 chip.

{"Time":"2024-07-26T09:20:59","ANALOG":{"Temperature1":28.8},"TempUnit":"C"}
{"Time":"2024-07-26T09:21:59","ANALOG":{"Temperature1":46.3},"TempUnit":"C"}
{"Time":"2024-07-26T09:22:59","ANALOG":{"Temperature1":50.9},"TempUnit":"C"}
{"Time":"2024-07-26T09:23:59","ANALOG":{"Temperature1":54.1},"TempUnit":"C"}
{"Time":"2024-07-26T09:24:59","ANALOG":{"Temperature1":56.9},"TempUnit":"C"}
{"Time":"2024-07-26T09:25:59","ANALOG":{"Temperature1":58.8},"TempUnit":"C"}
{"Time":"2024-07-26T09:26:59","ANALOG":{"Temperature1":60.6},"TempUnit":"C"}
{"Time":"2024-07-26T09:27:59","ANALOG":{"Temperature1":62.0},"TempUnit":"C"}
{"Time":"2024-07-26T09:28:59","ANALOG":{"Temperature1":63.2},"TempUnit":"C"}
{"Time":"2024-07-26T09:29:59","ANALOG":{"Temperature1":63.8},"TempUnit":"C"}
{"Time":"2024-07-26T09:30:59","ANALOG":{"Temperature1":64.7},"TempUnit":"C"}

For me this case is closed.

NOTE: If you want lower values if powered then change the B-constant of adcparam like ``adcparam1 2,10000,10000,6700`` ...

**Comment by dippes** [comment]
Thanks for the explanations. If the sensor is installed next to the chip, the high temperature is plausible.
What should the sensor be good for then?

**Reply by sfromis** [reply]
As I already mentioned earlier, "it is meant for protection against overheating". It is related to `SetOption42` for the threshold, but only with energy monitoring.
